pkg/policy/validate.go

```diff
@@ -413,6 +418,19 @@ func hasVariables(policy *kyverno.ClusterPolicy) [][]string {
 	return matches
 }
 
+func validateJMESPath(policy *kyverno.ClusterPolicy) bool {
+	for _, rule := range policy.Spec.Rules {
+		for _, context := range rule.Context {
+			if context.APICall.JMESPath == "jmesPath" {
```

Comment:
I think I've mentioned this earlier, but this is not what the issue describes. You're supposed to check whether **_all_** the opening braces are **_properly balanced_** with their corresponding ending braces within **_any_** variable expression.
 You should probably look around here:
https://github.com/kyverno/kyverno/blob/9ec53af2959946c2e6bf20061681c6ca7637020c/pkg/engine/variables/vars.go#L35-L39